crates/core/src/client/mod.rs

```diff
@@ -154,19 +152,18 @@ impl<T: JsonRpcTransport + Send + Sync> KakarotEthApi<T> for KakarotClient<JsonR
         let segmented_result = decode_eth_call_return(&call_result)?;
 
         // Convert the result of the function call to a vector of bytes
-        let return_data = segmented_result.first().ok_or_else(|| {
-            EthApiError::<T::Error>::OtherError(anyhow::anyhow!(
-                "Cannot parse and decode last argument of Kakarot call",
-            ))
-        })?;
-        if let FeltOrFeltArray::FeltArray(felt_array) = return_data {
-            let result: Vec<u8> = felt_array.iter().filter_map(|f| u8::try_from(*f).ok()).collect();
-            let bytes_result = Bytes::from(result);
-            return Ok(bytes_result);
+        let return_data = segmented_result
+            .first()
+            .ok_or_else(|| DataDecodingError::InvalidReturnArrayLength("return data".into()))?;
```

Comment:
Can we find something a bit more explicit than `return data`?
Like `cannot access return data array's first element`?

Reply:
Right now the error is defined as:
```
#[error("invalid call return length: failed on {0}")]
InvalidReturnArrayLength(String),
```
So for example the above will print "invalid call return length: failed on return data". The idea was that when decoding for example for `send_eth_transaction` (which has now been deleted), it allowed you to pin point what failed. As stated above, I got revert and do
```
#[error("invalid return array length, expected {expected}, got {actual}"]
DataDecodingError::InvalidReturnArrayLength{expected: usize, actual: usize}
```